hpx/runtime/components/server/create_component.hpp

```
@@ -79,8 +87,16 @@ namespace hpx { namespace components { namespace server
     naming::gid_type create(naming::gid_type const& gid,
         util::unique_function_nonser<void(void*)> const& ctor)
     {
-        Component* c = (Component*)Component::heap_type::alloc(1);
-        ctor(c);
+        void * cv = Component::heap_type::alloc(1);
+        try {
+            ctor(cv);
+        }
+        catch(...)
+        {
+            Component::heap_type::free(cv, 1);
+            return naming::invalid_gid;
+        }
+        Component *c = static_cast<Component *>(cv);
```

Same as above, please let the function throw on error.